### ISSUE TITLE
Retrieving trajectory for handwritten lines

### ISSUE THREAD

**Issue by umang13** [AyanKumarBhunia/Self-Supervised-Learning-for-Sketch#3]
Hi,
This is great work, thanks for your efforts. I was trying to train the Image2Coordinate model on line level images for the IAMOnline database to retrieve corresponding ground truth strokes. In the experiments section of your paper you mention segmenting the line level data into word level data before training the model on them. Is there a reason this segmentation was performed? Can the model not attempt to directly vectorize the entire line image? What is the difference in a normal sketch image such as those your model is trained on and these line level images of IAMOnline?

Thanks,
Umang

**Comment by AyanKumarBhunia** [comment]
@umang13 Thanks for your interest in our work. 

Is there a reason this segmentation was performed? -- For simplification. We wanted to show a generic framework where we did not want to focus much on the difficulty due to the high aspect ratio of line images -- accordingly, you need to adapt the stride of deconvolution or upsampling layer. However, it is definitely worth trying.

**Comment by umang13** [comment]
Thanks for your response. I had some difficulty in understanding the training of your decoder for the Image2Coordinate part. It seems you concatenate the ground truth coordinates to the decoder input before passing it to the decoder. I understand teacher-forcing is used in sequence models to help with the model training, but the code seems to use such a flow irrespective of whether the teacher forcing flag is set or not. 
When use_teacher_forcing is true :
`decoder_input = torch.cat((encoded_feature.unsqueeze(1), target_coord[:, i_num, :].unsqueeze(1)), dim=-1)`
When it is false :
`decoder_input = torch.cat((encoded_feature.unsqueeze(1).repeat(1, batch['seq_len'].max(), 1), target_coord[:,:-1, :]), dim=-1)`

What am I missing here?

Thanks,
Umang
